proposals/1859_github-auth/README.md

```diff
@@ -0,0 +1,527 @@
+  which external identity provider is used for authentication should be transparent.
+
+### Specification / How it Solves the Use Cases
+
+#### New GitHubIdentityProvider resource
+
+Add a new CRD called GitHubIdentityProvider. Here is an example:
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: GitHubIdentityProvider
+metadata:
+  namespace: supervisor
+  name: github
+spec:
+  github_api:
+    # Required only for GitHub Enterprise Server (on-prem).
+    # Defaults to using the GitHub's public API.
+    host: my.example.com
+    # X.509 Certificate Authority (base64-encoded PEM bundle).
+    # If omitted, a default set of system roots will be trusted.
+    # Required only for GitHub Enterprise Server (on-prem), and only
+    # when its CA will not be trusted by the default system roots.
+    certificateAuthorityData: LS0tLS1CRUdJTiBDRV...0tLS0tCg==
+  claims:
+    # Which property of the GitHub user record shall determine
+    # the username in Kubernetes. Can be either "id", "login",
+    # or "login:id".
+    # Login names can only contain alphanumeric characters and
+    # non-repeating hyphens, and may not start or end with hyphens.
+    # GitHub's users are allowed to change their login name,
+    # although it is inconvenient. If a GitHub user changed
+    # their login name from "foo" to "bar", then a second user
+    # might change their name from "baz" to "foo" in order
+    # to take the old username of the first user. For this
+    # reason, it is not as safe to make authorization decisions
+    # based only on the login name.
+    # If desired, an admin could configure identity
+    # transformation expressions on a FederationDomain to
+    # further customize these usernames.
+    # Defaults to "login:id", which is the login name, followed
+    # by a colon, followed by the unique and unchanging integer
+    # ID number. This blends human-readable login names with the
+    # unchanging ID value. Note that colons are not allowed in
+    # GitHub login names or in ID numbers, so the colon in the
+    # "login:id" name will always be the login/id separator colon.
+    username: id
+    # Which property of the GitHub team record shall determine
+    # the group name in Kubernetes. Can be either "name" or "slug".
+    # Team names can contain upper and lower case characters,
+    # whitespace, and punctuation (e.g. "Kube admins!").
+    # Slug names are lower case alphanumeric characters and may
+    # contain dashes (e.g. "kube-admins").
+    # Either way, group names will always be prefixed by the
+    # GitHub org name followed by a colon (e.g. my-org:my-team).
+    # Note that colons are not allowed in GitHub org names,
+    # so the first colon in the group name will always be the
+    # org/team separator colon.
+    # If desired, an admin could configure identity
+    # transformation expressions on a FederationDomain to
+    # further customize these group names.
+    # Defaults to "slug".
+    groups: slug
```

Comment:
Yes, as noted in the comment above:

```
     # Note that colons are not allowed in GitHub org names,
     # so the first colon in the group name will always be the
     # org/team separator colon.
```

We could also consider using `/` instead of `:` because you found that's what the GitHub GraphQL API does when you use `combinedSlug`. Although note that Dex uses `:` so some people will already be used to that.